src/messages/IPostMessageSentHandler.ts

```diff
@@ -1,23 +1,25 @@
-import { IHttp } from '../accessors/IHttp';
-import { IRead } from '../accessors/index';
+import { IHttp, IRead } from '../accessors/index';
 import { IMessage } from './IMessage';
 
 export interface IPostMessageSentHandler {
     /**
-     * First step when a handler is executed: Enables the handler to signal
-     * to the Rocketlets framework whether handler shall actually execute for the message
-     * about to be sent.
+     * Enables the handler to signal to the Rocketlets framework whether
+     * this handler should actually be executed for after the message
+     * has been sent.
      *
-     * @param message The message which is being sent
+     * @param message The message which was sent
      * @param read An accessor to the environment
-     * @return true: run the pre-logic
+     * @param http An accessor to the outside world
+     * @return whether to run the executor function
      */
-    checkPostMessageSent(message: IMessage, read: IRead): boolean;
+    checkPostMessageSent?(message: IMessage, read: IRead, http: IHttp): boolean;
 
     /**
      * Method called *after* the message is sent to the other clients.
      *
      * @param message The message which was sent
+     * @param read An accessor to the environment
+     * @param http An accessor to the outside world
      */
-    communicatePostMessageSent(message: IMessage, read: IRead, http: IHttp): void;
```

Comment:
@graywolf336 I'm fine with having a bit more generic method here.
What I believe should be in an own method (or even interface) were all "disruptive" changes, such as modifying existing model data.

For the installation, it could be distinguished based on the implemented:

> The plugin performs the following actions:
> - It reacts on created messages (e. g. by sending data to a remote system or by persisting additional data (in case of `executePostMessageSent`)
> - It alters data which has been stored within Rocket.Chat (e. g. changes a room's members) (in case of `modifyPostMessageSent`)